File: app/src/main/java/com/kelsos/mbrc/ui/fragments/HelpFragment.java
package com.kelsos.mbrc.ui.fragments;

import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.kelsos.mbrc.R;
import com.kelsos.mbrc.utilities.RemoteUtils;
import timber.log.Timber;

public class HelpFragment extends Fragment {

  @BindView(R.id.help_webview)
  WebView helpWebview;

  public HelpFragment() {
    // Required empty public constructor
  }

  @NonNull
  public static HelpFragment newInstance() {
    return new HelpFragment();
  }

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
  }

  @Override
  public void onStart() {
    super.onStart();

    String url;
    try {
      url = String.format("http://kelsos.net/musicbeeremote/help?version=%s", RemoteUtils.getVersion(getContext()));
    } catch (PackageManager.NameNotFoundException e) {
      Timber.v(e, "Failed to get version");
      url = "http://kelsos.net/musicbeeremote/help";
    }
    helpWebview.loadUrl(url);
  }

  @Override
  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    // Inflate the layout for this fragment
    View view = inflater.inflate(R.layout.fragment_help, container, false);
    ButterKnife.bind(this, view);
    helpWebview.setWebViewClient(new RemoteWebViewClient());
    return view;
  }

  private static class RemoteWebViewClient extends WebViewClient {
    @Override
    public boolean shouldOverrideUrlLoading(WebView view, String url) {
      view.loadUrl(url);
      return false;
    }
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/ui/help_feedback/HelpFragment.kt
package com.kelsos.mbrc.ui.help_feedback

import android.content.pm.PackageManager
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebView
import android.webkit.WebViewClient
import butterknife.BindView
import butterknife.ButterKnife
import com.kelsos.mbrc.R
import com.kelsos.mbrc.utilities.RemoteUtils
import timber.log.Timber

class HelpFragment : Fragment() {

  @BindView(R.id.help_webview) lateinit var helpWebview: WebView

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
  }

  override fun onStart() {
    super.onStart()

    val url: String
    url = try {
      String.format("https://mbrc.kelsos.net/help?version=%s", RemoteUtils.getVersion(requireContext()))
    } catch (e: PackageManager.NameNotFoundException) {
      Timber.v(e, "Failed to get version")
      "https://mbrc.kelsos.net/help"
    }

    helpWebview.loadUrl(url)
  }

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
    // Inflate the layout for this fragment
    val view = inflater.inflate(R.layout.fragment_help, container, false)
    ButterKnife.bind(this, view)
    helpWebview.webViewClient = RemoteWebViewClient()
    return view
  }

  private class RemoteWebViewClient : WebViewClient() {
    @Suppress("OverridingDeprecatedMember")
    override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
      view.loadUrl(url)
      return false
    }
  }

  companion object {

    fun newInstance(): HelpFragment {
      return HelpFragment()
    }
  }
}// Required empty public constructor
